medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -63,15 +64,33 @@ impl Error for RetrieveFromQueueError {
     }
 }
 
+pub struct RetrievedMessage<T> {
+    pub message: T,
+    peek_lock: PeekLockResponse,
+}
+
+impl<T> RetrievedMessage<T> {
+    pub async fn remove(&self) -> Result<String, anyhow::Error> {
+        let queue_removal_result = self.peek_lock.delete_message().await.map_err(|e| {
+            tracing::error!("{:?}", e);
+            anyhow!("Queue Removal Error")
+        });
+        tracing::info!("Removed job from ServiceBus ({:?})", queue_removal_result);
```

Comment:
With tracing, will we automatically get the context of the Job ID of the message that was removed from the queue?

Reply:
not here, we don't have access to it at this point and don't know that `T` even has a `job_id` - we could do it higher up where we make the call to this `remove` method